sample/marchingSquares.glsl

```diff
@@ -0,0 +1,204 @@
+#include "../generative/pnoise.glsl"
+#include "../math/const.glsl"
+#include "../math/rotate2d.glsl"
+#include "../sample.glsl"
+
+/*
+Author: Guido Schmidt
+description: The Marching Squares algorithm generates contour lines (isolines) of a given input for a
+given grid and a threshhold. Is uses a lookup of 16 cases to decide if a cell is inside, outside or
+a specific corner.
+use: <vec3> marchingSquares(in <vec2> uv) 
+options:
+    - SAMPLE_MARCHING_SQUARES_FNC(UV): optional sampling function, defaults so sampling of pnoise
+*/
+
+#ifndef SAMPLE_MARCHING_SQUARES_FNC
+float defaultSampleFunction(in vec2 uv) {
+    float dv = pnoise(uv * 5.0, vec2(0.0));
+    return dv;
+}
+#define SAMPLE_MARCHING_SQUARES_FNC(UV) defaultSampleFunction(UV)
+#endif
+
+#define centerOf(p1, p2, v1, v2) mix(p1, p2, 0.5)
+
+float sampleOutline(in vec2 p, in vec2 cellUv, in vec2 a, in vec2 b, in bool straight) {
```

Comment:
this can potentially produce name collisions. 
Let's put the name of the main function first like
```
float sampleMarchingSquares_outline(...) {
}
```